SCClassLibrary/Common/Collections/Event.sc

```diff
@@ -44,17 +47,9 @@ Event : Environment {
 		//		^this.delta
 	}
 
-	// this[\isRest] may be nil
 	isRest {
-		^this[\isRest] == true
-		or: { this[\type] == \rest
-			or: {
-				this.use {
-					parent ?? { parent = defaultParentEvent };
-					~detunedFreq.value.isRest
-				}
-			}
-		}
+		_Event_IsRest
+		^false
```

Comment:
If we need to add new possible rest cases in the future, as the code is here, you can simply add them to the method, instead of adding them to the method *and* changing the primitive code. I had raised the question earlier about raw speed now vs future extensibility later. I lean toward extensibility.